arc

```diff
@@ -0,0 +1,113 @@
+#!/bin/bash
+
+set -e
+
+usage () {
+  echo "Usage: $0 [OPTIONS] <INPUT-FILE> [-- <MLIR-OPTS>]"
+  echo "Options:"
+  echo "  -h       Show this help message"
+  echo "  -o <DIR> Output directory (defaults to ./target/)"
+  exit 0
+}
+
+while getopts ho: opt; do
+    case $opt in
+        h)
+            usage
+            ;;
+        o)
+            CRATE_DIR=$OPTARG
+            ;;
+        \?)
+            echo "Invalid option: -$OPTARG" >&2
+            usage
+            exit 1
+            ;;
+    esac
+done
+
+while [[ $# -gt 0 ]]; do
+    case $1 in
+        --)
+            shift
+            break
+            ;;
+        *)
+            INPUT_FILE=$1
+            ;;
+    esac
+    shift
+done
+
+CRATE_NAME=$(basename $INPUT_FILE)-crate
+
+if [ -z $INPUT_FILE ]; then
+    usage
+fi
+
+if [ -z $CRATE_DIR ]; then
+    CRATE_DIR=$(dirname $INPUT_FILE)/$CRATE_NAME
+fi
+
+if [ ! -x $(which arc-lang) ]; then
+    echo "arc-lang not found. Please run 'make' in the root directory."
+    exit 1
+fi
+
+if [ ! -x $(which arc-mlir) ]; then
+    echo "arc-mlir not found. Please run 'make' in the root directory."
+    exit 1
+fi
+
+if [ ! -f $ARC_RUNTIME_DIR ]; then
```

Comment:
I can't see that `ARC_RUNTIME_DIR` is set by the script, is this intentional?

If our build system installs `arc`, that should be enough to find the path to the other binaries. It should be enough to give a full path to `arc` for the other binaries to be found, i.e. `/foo/bar/bin/arc` should be enough.